test/integration/sensing.js

```diff
@@ -0,0 +1,35 @@
+var path = require('path');
+var test = require('tap').test;
+var extract = require('../fixtures/extract');
+var VirtualMachine = require('../../src/index');
+
+var uri = path.resolve(__dirname, '../fixtures/sensing.sb2');
+var project = extract(uri);
+
+test('sensing', function (t) {
```

Comment:
I'm getting the feeling that this duplicated code (below) should be in one shared place so that these basic project tests can `require` the common code. Ideally they could also add tests of their own to be called from the `playgroundData` event handler. It looks to me like the `threads.length` check and the project URI, and maybe the timeout duration, are the only test-specific parts of this right now.

I don't think it's critical to do this at the moment... if you don't object to such a refactor then I can file a low-priority issue for it.